go/ql/lib/semmle/go/frameworks/AwsLambda.qll

```diff
@@ -0,0 +1,59 @@
+/**
+ * Provides classes for working with untrusted flow sources, sinks and taint propagators
+ * from the `github.com/aws/aws-lambda-go/lambda` package.
+ */
+
+import go
+
+/** A source of input data in an AWS Lambda. */
+private class LambdaInput extends UntrustedFlowSource::Range {
+  LambdaInput() {
+    exists(Parameter p | p = this.asParameter() |
+      p = any(HandlerFunction hf).getAParameter() and
+      not p.getType().hasQualifiedName("context", "Context") and
+      not p instanceof ReceiverVariable
+    )
+  }
+}
+
+private class HandlerFunction extends FuncDef {
+  HandlerFunction() {
+    exists(StartOrNewHandlerFunction f, Expr e |
+      f.getACall().getArgument(f.getHandlerArgPos()).asExpr() = e
+    |
+      this = e.(FunctionName).getTarget().getFuncDecl() or
+      this = e.(FuncLit)
```

Comment:
```suggestion
    exists(StartOrNewHandlerFunction f, DataFlow::Node handlerArg |
      f.getACall().getArgument(f.getHandlerArgPos()) = handlerArg
    |
      handlerArg = this.(FuncDecl).getFunction().getARead() or
      handlerArg = DataFlow::exprNode(this.(FuncLit))
```

Slightly nicer way to phrase this, and more robust to different ways to refer to a declared function.